src/services/tradeApiClient.js

```diff
@@ -336,32 +345,44 @@ class TradeApiClient {
    * Process API HTTP request.
    *
    * @param {string} endpointPath API endpoint path and action.
-   * @param {Object} payload Request payload parameters object.
+   * @param {*} payload Request payload parameters object.
    * @param {string} [method] Request HTTP method, currently used only for cache purposes.
    * @param {string} [token] Optional authentication token.
    * @returns {Promise<*>} Promise that resolves Trade API request response.
    *
    * @memberof TradeApiClient
    */
-  async doRequest(endpointPath, payload, method = "", token) {
-    const requestUrl = this.baseUrl + endpointPath;
+  async doRequest(endpointPath, payload, method = "POST", token) {
+    const baseUrl =
+      payload.apiVersion && payload.apiVersion === 2 ? this.baseUrlv2 : this.baseUrlv1;
+    let requestUrl = baseUrl + endpointPath;
     let responseData = {};
 
     const authToken = this.token || token;
 
+    /**
+     * @type {*}
+     */
+    let options = {
+      method: method,
+      body: null,
+      headers: {
+        "Content-Type": "application/json",
+        "X-API-KEY": process.env.GATSBY_API_KEY || "",
+        ...(authToken && { Authorization: "Bearer " + authToken }),
+      },
+    };
+
+    if (method === "GET") {
+      requestUrl = payload.query
+        ? `${requestUrl}?${new URLSearchParams(payload.query)}`
```

Comment:
Instead of passing a `query` property in `payload`, just use `URLSearchParams(payload)` (GET)

Reply:
Done